physicsnemo/distributed/shard_utils/point_cloud_ops.py

```diff
@@ -532,111 +538,124 @@ def ball_query_layer_wrapper(
     like in the non-sharded case.
 
     Args:
         func: Original forward method
         type: Types of the inputs
-        args: Positional arguments (points1, points2)
+        args: Positional arguments (points, queries)
         kwargs: Keyword arguments
 
     Returns:
         Tuple of (mapping, num_neighbors, outputs) as torch.Tensor or ShardTensor
     """
 
-    points1, points2, bq_kwargs = repackage_ball_query_args(*args, **kwargs)
+    points, queries, bq_kwargs = repackage_radius_search_wrapper_args(*args, **kwargs)
+
+    if bq_kwargs["max_points"] is None:
+        raise MissingShardPatch(
+            "sharded radius_search_wrapper for radius_search does not currently support max_points=None"
+        )
+
+    if bq_kwargs["return_dists"] is True:
+        raise MissingShardPatch(
+            "sharded radius_search_wrapper for radius_search does not currently support return_dists=True"
+        )
 
     # Make sure all meshes are the same
-    if points1._spec.mesh != points2._spec.mesh:
+    if points._spec.mesh != queries._spec.mesh:
         raise MissingShardPatch(
-            "point_cloud_ops.ball_query_layer_wrapper: All point inputs must be on the same mesh"
+            "point_cloud_ops.radius_search_wrapper: All point inputs must be on the same mesh"
         )
 
     # make sure all meshes are 1D
-    if points1._spec.mesh.ndim != 1:
+    if points._spec.mesh.ndim != 1:
         raise MissingShardPatch(
-            "point_cloud_ops.ball_query_layer_wrapper: All point inputs must be on 1D meshes"
+            "point_cloud_ops.radius_search_wrapper: All point inputs must be on 1D meshes"
         )
 
     # Do we need a ring?
-    points2_placement = points2._spec.placements[0]
-    if points2_placement.is_shard():
+    points_placement = points._spec.placements[0]
+
+    if points_placement.is_shard():
         # We need a ring
-        mapping, num_neighbors, outputs = ring_ball_query(points1, points2, bq_kwargs)
+        mapping, outputs, _, num_neighbors = ring_ball_query(points, queries, bq_kwargs)
     else:
         # No ring is needed
         mapping, num_neighbors, outputs = ringless_ball_query(
-            points1, points2, bq_kwargs
+            points, queries, bq_kwargs
         )
 
-    return mapping, num_neighbors, outputs
+    return mapping, outputs, None, num_neighbors
 
 
-def repackage_ball_query_args(
-    points1: Union[torch.Tensor, ShardTensor],
-    points2: Union[torch.Tensor, ShardTensor],
-    k: int,
+def repackage_radius_search_wrapper_args(
+    points: torch.Tensor | ShardTensor,
+    queries: torch.Tensor | ShardTensor,
     radius: float,
-    hash_grid: wp.HashGrid,
-    *args: Any,
-    **kwargs: Any,
+    max_points: int | None = None,
+    return_dists: bool = False,
+    return_points: bool = False,
+    *args,
+    **kwargs,
 ) -> Tuple[
-    Union[torch.Tensor, ShardTensor],
-    Union[torch.Tensor, ShardTensor],
     Union[torch.Tensor, ShardTensor],
```

Comment:
This doesn't have to happen in this PR, but given how common this overloaded type hint is, it might be worth refactoring these onto a custom type definition:
```python
TensorLike = torch.Tensor | ShardTensor
```

And then later we can just use `TensorLike` instead of constantly doing these unions (which, by the way, can be done with `|` instead of `Union` in Python 3.10+)

Reply:
Actually, some of this is legacy cruft from the previous way of handling function wrappers.  These ops should only ever ingest and return `ShardTensor` now, which makes type hints easier!